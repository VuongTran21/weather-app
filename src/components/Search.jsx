import { useCallback, useEffect, useState } from 'react';
import {
  Form,
  Spinner,
} from 'react-bootstrap';
import { useForecast } from '../hooks/useForecast';
import debounce from 'lodash.debounce';

export default function Search({ onForecastChange }) {
  const [inputText, setInputText] = useState('');
  const [{ loading, forecasts }, getForecasts] = useForecast();
  useEffect(() => {
    if (inputText) {
      getForecasts(inputText);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    onForecastChange(forecasts);
  }, [forecasts]);

  const handleSearchChangeDebounce = useCallback(
    debounce((nextValue) => getForecasts(nextValue), 1000),
    []
  );

  const onChange = (e) => {
    const text = e.target.value;
    setInputText(text);

    handleSearchChangeDebounce(text);
  };

  return (
    <Form.Group controlId="cityName">
      <div className="search-wrapper">
        <Form.Control
          data-testid="input-search"
          onChange={onChange}
          value={inputText}
          type="text"
          placeholder="Enter your city name"
        />

        {loading && (
          <Spinner
            data-testid="loading"
            as="span"
            animation="border"
            size="sm"
            role="status"
            aria-hidden="true"
          />
        )}
      </div>
    </Form.Group>
  );
}
