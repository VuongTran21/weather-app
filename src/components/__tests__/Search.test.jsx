import React from 'react';
import { render } from '@testing-library/react';

import Search from '../Search';

describe('Search Component', () => {
  it('should match snapshot', () => {
    const { container } = render(<Search onForecastChange={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });
});
