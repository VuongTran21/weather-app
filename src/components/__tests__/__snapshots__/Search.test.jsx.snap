// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should match snapshot 1`] = `
<div>
  <div
    class="form-group"
  >
    <div
      class="search-wrapper"
    >
      <input
        class="form-control"
        data-testid="input-search"
        id="cityName"
        placeholder="Enter your city name"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
